src/test/java/QueryInformationTest.java

```diff
@@ -0,0 +1,146 @@
+
+@RunWith(JUnit4.class)
+public final class QueryInformationTest {
+  private static final String EMPTY_STRING = "";
+  private static final String NULL_VALUE = null;
+  private static final ImmutableSet<String> EMPTY_IMMUTABLE_SET = ImmutableSet.of();
+  private static final double ERROR_THRESHOLD = 0.1;
+
+  // Values for a valid test.
+  private static final String CST_TIMEZONE_ID = "America/Chicago";
+  private static final String CATEGORIES = "Breakfast";
+  private static final String DATE_RANGE = "June 1, 2020 - June 30, 2020";
+  private static final String STORE = "McDonald's";
+  private static final String MIN_PRICE = "21.30";
+  private static final String MAX_PRICE = "87.60";
+
+  // Expected values.
+  private static final TimeZone CST_TIMEZONE = TimeZone.getTimeZone("America/Chicago");
+  private static final ImmutableSet EXPECTED_CATEGORIES = ImmutableSet.of("breakfast");
+  private static final long JUNE_1_2020_START_OF_DAY = 1590987600000L;
+  private static final long JUNE_30_2020_END_OF_DAY = 1593579599999L;
+  private static final String EXPECTED_STORE = "mcdonald's";
+  private static final double EXPECTED_MIN_PRICE = 21.30;
+  private static final double EXPECTED_MAX_PRICE = 87.60;
+
+  // Values for an invalid test.
+  private static final TimeZone GMT_TIMEZONE = TimeZone.getTimeZone("GMT");
+
+  // Edge case values.
+  private static final String EDGE_CASE_CATEGORIES = "   breAkfaSt ";
+  private static final String EDGE_CASE_STORE = "  MCdoNald'S";
+
+  @Test
+  public void validTimeZoneIdSetsToCst() throws ParseException {
+    QueryInformation queryInformation =
+        new QueryInformation(CST_TIMEZONE_ID, CATEGORIES, DATE_RANGE, STORE, MIN_PRICE, MAX_PRICE);
+    Assert.assertEquals(CST_TIMEZONE, queryInformation.getTimeZone());
+  }
+
+  @Test
+  public void invalidTimeZoneIdSetsToGmt() throws ParseException {
+    // Test with empty string.
+    QueryInformation queryInformation =
+        new QueryInformation(EMPTY_STRING, CATEGORIES, DATE_RANGE, STORE, MIN_PRICE, MAX_PRICE);
+    Assert.assertEquals(GMT_TIMEZONE, queryInformation.getTimeZone());
+  }
+
+  @Test
+  public void validCategories() throws ParseException {
+    QueryInformation queryInformation =
+        new QueryInformation(CST_TIMEZONE_ID, CATEGORIES, DATE_RANGE, STORE, MIN_PRICE, MAX_PRICE);
+    Assert.assertEquals(EXPECTED_CATEGORIES, queryInformation.getCategories());
+  }
+
+  @Test
+  public void categoriesLowercasedAndWhitespaceRemoved() throws ParseException {
+    QueryInformation queryInformation = new QueryInformation(
+        CST_TIMEZONE_ID, EDGE_CASE_CATEGORIES, DATE_RANGE, STORE, MIN_PRICE, MAX_PRICE);
+    Assert.assertEquals(EXPECTED_CATEGORIES, queryInformation.getCategories());
+  }
+
+  @Test
+  public void invalidCategories() throws ParseException {
+    // Test with empty string.
+    QueryInformation queryInformation = new QueryInformation(
+        CST_TIMEZONE_ID, EMPTY_STRING, DATE_RANGE, STORE, MIN_PRICE, MAX_PRICE);
```

Comment:
If parameter isn't self-explanatory, add a comment before it. Example:
QueryInformation(CST_TIMEZONE_ID, /* categories = */ EMPTY_STRING)